test/models/partner_test.rb

```diff
@@ -187,4 +187,30 @@ class PartnerTest < ActiveSupport::TestCase
     problems = partner.errors[:base]
     assert_equal 'Partners must have at least one of service area or address', problems.first
   end
+
+  test 'partner can have up to 3 "Category" tags' do
+    partner = create(:partner)
+    assert_predicate partner, :valid?
+
+    partner.tags << create(:tag, type: 'Category', name: 'Category Tag 1')
```

Comment:
just an fyi this is a rare case you can use ruby `time` which you learn in the first day you learn ruby then never get to touch again lol https://www.dotnetperls.com/iterator-ruby

Reply:
i used `x.times do` just today in some tests.